InvenTree/order/models.py

```diff
@@ -1602,7 +1602,16 @@ def clean(self):
 
         try:
             if self.line.part != self.item.part:
-                errors['item'] = _('Cannot allocate stock item to a line with a different part')
+                # Check for variant_of mapping on item.part to see if this trees up to line.part
```

Comment:
@1337joe there is a much more efficient way to make this query (rather than the while loop approach) - although there is no reason that you should have known this :)

```python
variants = self.line.part.get_descendants(include_self=True)

if self.item.part in variants:
    # Good to go!
else:
    # Not a direct match *or* a variant
```

The `get_descendants` method gets all variants (all the way down the tree) in a single db query